lib/mix/tasks/dicon.control.ex

```diff
@@ -39,9 +40,13 @@ defmodule Mix.Tasks.Dicon.Control do
     end
   end
 
-  defp exec(conn, target_dir, command) do
+  defp exec(conn, host_config, target_dir, command) do
     otp_app = config(:otp_app) |> Atom.to_string
-    command = [target_dir, "/current/bin/", otp_app, ?\s, command]
+    env =
+      Enum.map(host_config[:os_env] || %{}, fn({key, value}) ->
+        [key, ?=, inspect(to_string(value)), ?\s]
```

Comment:
@lexmag I did `inspect(to_string(...))` so that we get double-quotes escaping for free, got any better ideas/can you see any problems with this?